packages/zowe-explorer-api/__tests__/__unit__/tree/pagination.unit.test.ts

```diff
@@ -0,0 +1,615 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { Paginator, FetchFn, IFetchResult, NavigationTreeItem } from "../../../src";
+import * as vscode from "vscode";
+import { MockedProperty } from "../../../__mocks__/mockUtils";
+
+// Helper type for mock data items
+type MockDataItem = { id: number; name: string };
+
+// Helper function to create mock data
+const createMockData = (count: number, prefix = "item-"): MockDataItem[] => {
+    return Array.from({ length: count }, (_, i) => ({ id: i, name: `${prefix}${i}` }));
+};
+
+// Interface for mock function options
+interface MockFetchOptions {
+    failOnCursor?: string | undefined;
+    returnLessItemsOnCursor?: string | undefined;
+}
+
+const createMockFetchFunction = (totalItems: number, options?: MockFetchOptions): FetchFn<MockDataItem, string> => {
+    const failOnCursor = options?.failOnCursor;
+    const returnLessItemsOnCursor = options?.returnLessItemsOnCursor;
+    const failOnCursorProvided = !!options && Object.prototype.hasOwnProperty.call(options, "failOnCursor");
+
+    return jest.fn(async (cursor: string | undefined, limit: number): Promise<IFetchResult<MockDataItem, string>> => {
```

Comment:
a few lint warnings in regards to `async` functions without any `await` statements, but that's kind of a minor thing 😅 

Could either disable that lint rule for these or return a `Promise.resolve` 😋

Reply:
Sorry about that, addressed in b61fca9